main.tf

```diff
@@ -133,3 +133,14 @@ resource "aws_apigatewayv2_integration" "this" {
     ignore_changes = [passthrough_behavior]
   }
 }
+
+# VPC Link (Private API)
+resource "aws_apigatewayv2_vpc_link" "this" {
+  count = var.create && length(var.security_group_ids) > 0 && length(var.subnet_ids) > 0 ? 1 : 0
```

Comment:
I would recommend to rewrite this to avoid using variables which can be unknown during plan time and introduce another boolean variable like `create_vpc_link`.

This way users will be able to create VPC and pass outputs from the module instead of just relying on data-sources (as you do in the example).

PS: Could you also update example to use VPC module for real?